rad/src/lib.rs

```diff
@@ -245,11 +258,20 @@ pub fn run_aggregation_with_context_report(
 ) -> Result<RadonReport<RadonTypes>> {
     let filters = aggregate.filters.as_slice();
     let reducer = aggregate.reducer;
-    let radon_script = create_radon_script_from_filters_and_reducer(filters, reducer)?;
+    let is_rng = reducer == 0x11;
 
-    let items_to_aggregate = RadonTypes::from(RadonArray::from(radon_types_vec));
+    if is_rng {
+        Ok(RadonReport::from_result(
+            Ok(radon_types_vec[0].clone()),
+            context,
+        ))
+    } else {
```

Comment:
What is this doing?

Reply:
unwrap() :P

Reply:
In the rng case the request will usually have only one retrieval source. So this way we avoid the aggregation step, which is not needed. However I think that could apply to all kinds of requests, because you don't need aggregation if there is only one source. This would be a breaking change, but we can implement it with TAPI.

Reply:
I removed it and use the Mode instead, because the mode of a vector of one element is that element. To avoid strange cases